src/traversals/bfs.jl

```diff
@@ -65,23 +65,26 @@ For vertices in disconnected components the default distance is -1.
 function gdistances!(g::AbstractGraph, source, dists)
     T = eltype(g)
     n = nv(g)
-    fill!(dists, -1)
+    fill!(dists, typemax(T))
```

Comment:
Why switch from -1 indications to typemax(T)?

Reply:
Oh, because we might be using unsigned ints.